File: PRG_410/WEEK_4/Assignment/code_Java/AssignmentWeek4/Triangle.java

import java.util.*;

/**
 * 
 */
public class Triangle implements IShape {

    /**
     * Default constructor
     */
    public Triangle() {
    }

    /**
     * 
     */
    private double mWidth;

    /**
     * 
     */
    private double mHeight;

    /**
     * @return
     */
    public double Area() {
        // TODO implement here
        return 0.0d;
    }

    /**
     * @return
     */
    public double Perimeter() {
        // TODO implement here
        return 0.0d;
    }

    /**
     * @param width 
     * @param height
     */
    public void Triangle(double width, double height) {
        // TODO implement here
    }

    /**
     * 
     */
    private void Triangle() {
        // TODO implement here
    }

    /**
     * @return
     */
    public String Type() {
        // TODO implement here
        return "";
    }

    /**
     * @return
     */
    abstract double Area();

    /**
     * @return
     */
    public abstract double Perimeter();

    /**
     * @return
     */
    public abstract String Type();

}

File: PRG_410/WEEK_4/AssignmentWeek4/AssignmentWeek4/Triangle.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

public class Triangle : IShape {

    private Triangle() {
    }

    private double mWidth;

    private double mHeight;

    /// <summary>
    /// @return
    /// </summary>
    public double Area() {
        // TODO implement here
        return mWidth * mHeight / 2;
    }

    /// <summary>
    /// @return
    /// </summary>
    public double Perimeter() {
        // TODO implement here
        double diagSq = mHeight * mHeight + mWidth * mWidth;
        double diag = Math.Sqrt(diagSq);
        return ( diag + mHeight + mWidth );
    }

    /// <summary>
    /// @param width 
    /// @param height
    /// </summary>
    public Triangle( Width width, Height height) {
        mWidth = width.Value;
        mHeight = height.Value;
    }


    /// <summary>
    /// @return
    /// </summary>
    public String Type() {
        // TODO implement here
        return "Right Triangle";
    }

}